pkg/controller/upload_controller.go

```diff
@@ -105,13 +105,22 @@ func (c *uploadController) enqueueUploadItem(obj interface{}) {
 
 	switch req.Status.Phase {
 	case "", pluginv1api.UploadPhaseNew, pluginv1api.UploadPhaseInProgress, pluginv1api.UploadPhaseUploadError, pluginv1api.UploadPhaseCanceling:
-		// Process New and InProgress and UploadError Uploads
-	case pluginv1api.UploadPhaseCanceled:
-		// The upload was canceled, nothing to do.
-		log.Debug("The upload request was canceled")
-		return
+		// Process New and InProgress and UploadError and UploadCanceling Uploads
 	default:
-		log.Debug("Upload CR is not New or InProgress or UploadError, skipping")
+		log.Debug("Upload CR is not New or InProgress or UploadError or UploadCanceling, skipping")
+		if req.Status.Phase == pluginv1api.UploadPhaseCanceled {
```

Comment:
same as before, use the case statements to explicitly specify phases on which delete occurs. Perhaps move the delete to different method as well, looks clean.

Reply:
Same as above.